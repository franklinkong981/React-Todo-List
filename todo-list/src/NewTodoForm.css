.NewTodoForm-input-label {
  margin-right: 10px;
}

.NewTodoForm-todo-input {
  font-size: 24px;
}

.NewTodoForm-submit-button {
  background-color: green;
  color: white;
  margin-left: 10px;
}