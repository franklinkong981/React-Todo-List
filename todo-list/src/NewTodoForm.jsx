import react from "react";
import {useFormik} from 'formik';

const NewTodoForm = ({addTodoFunction}) => {
  const formik = useFormik({
    initialValues:{
      content: ''
    },
    onSubmit: values => {
      addTodoFunction(values.content);
    }
  });

  return (
    <form className="NewTodoForm" onSubmit={formik.handleSubmit}>
        <label htmlFor="todo-input-field">Enter New Todo:</label>
        <input id="todo-input-field" className="NewTodoForm-todo-input" type="text" name="content"
        value={formik.values.content} onChange={formik.handleChange}/>
        <button type="submit">Add New Todo</button>
    </form>
  );
};

export default NewTodoForm;