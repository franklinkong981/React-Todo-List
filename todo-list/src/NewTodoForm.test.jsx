import React from "react";
import {render} from "@testing-library/react";
import NewTodoForm from "./NewTodoForm.jsx";

const addTodo = () => {
  console.log("Todo added!");
}

it("renders without crashing", function() {
  render(<NewTodoForm addTodoFunction={addTodo}/>);
});

it("matches snapshot", function() {
  const {asFragment} = render(<NewTodoForm addTodoFunction={addTodo}/>);
  expect(asFragment()).toMatchSnapshot();
});