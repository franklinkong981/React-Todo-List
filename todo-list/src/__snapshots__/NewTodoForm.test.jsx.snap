// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewTodoForm"
  >
    <label
      class="NewTodoForm-input-label"
      for="todo-input-field"
    >
      Enter New Todo:
    </label>
    <input
      class="NewTodoForm-todo-input"
      id="todo-input-field"
      name="content"
      size="50"
      type="text"
      value=""
    />
    <button
      class="NewTodoForm-submit-button"
      type="submit"
    >
      Add New Todo
    </button>
  </form>
</DocumentFragment>
`;
